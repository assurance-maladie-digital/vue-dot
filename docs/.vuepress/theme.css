html, body {
	font-size: 100% !important;
}

.v-ripple__container,
.v-ripple__container * {
	max-width: none;
}

.v-input--selection-controls__ripple {
	max-width: none;
}

.application--wrap {
	width: 100%;
}

.theme--light.v-input--is-disabled .v-label,
.theme--light.v-icon.v-icon--disabled,
.theme--light.v-list .v-list--disabled {
	color: #767676 !important;
}

.theme--light.v-btn.v-btn--disabled .v-icon {
	color: #949494 !important;
}

.v-tabs__bar .v-tabs__item--disabled {
	color: #333 !important;
}

.theme--light.v-btn.v-btn--disabled {
	color: #646464 !important;
}

.content h1 {
	font-size: 2.2rem;
	margin-bottom: 25px;
}

.content h2 {
	font-size: 1.65rem;
	margin-bottom: 16px;
	margin-top: 35px;
}

.content h2 + h3 {
	margin-top: 0;
}

.content h3 {
	font-size: 1.35rem;
	margin-bottom: 10px;
	margin-top: 35px;
}

.content h3 + h4 {
	margin-top: 0;
}

.content h4 {
	font-size: 1.15rem;
	margin-bottom: 5px;
	margin-top: 15px;
}

pre[class*='language-'] {
	align-items: center;
	box-shadow: none;
	display: flex;
	border-radius: 2px;
	position: relative;
	overflow-x: auto;
	overflow-y: hidden;
	background: #2d2d2d;
	margin: 0;
	margin-bottom: 5px;
}

code[class*='language-'], code {
	position: relative;
	box-shadow: none;
	overflow-x: auto;
	overflow-y: hidden;
	word-break: break-word;
	flex-wrap: wrap;
	align-items: center;
	vertical-align: middle;
	white-space: pre-wrap;
	margin: 0;
	font-family: 'Inconsolata', monospace;
	font-weight: 300;
	font-size: .9rem;
	line-height: 1.55;
	padding: 0 5px;
}

code {
	background: rgba(240, 240, 240, .8);
	color: rgba(0, 0, 0, .7);
}

code::after,
code::before {
	content: none;
}

pre code {
	background: transparent;
	padding: 0;
	color: #ccc;
}

.header-anchor {
	color: #0c4887;
	transition: opacity .15s !important;
	text-decoration: none;
	font-size: .85em;
	float: left;
	margin-left: -.8em;
	margin-top: .125em;
	padding-right: .23em;
	opacity: 0;
	position: absolute;
}

[id]:hover > .header-anchor,
.header-anchor:focus {
	opacity: 1;
}

.icon.outbound {
	color: inherit;
}

.custom-block {
	padding: .1rem 1.5rem;
	border-left-width: .5rem;
	border-left-style: solid;
	margin: 1rem 0;
	background-color: #f3f5f7 !important;
	border-color: #42b983;
}

.custom-block.warning {
	background-color: rgba(255, 229, 100, .3) !important;
	border-color: #e7c000 !important;
}

.custom-block .custom-block-title {
	font-weight: 600;
	margin-top: 10px;
	margin-bottom: 0;
}

.custom-block.warning .custom-block-title {
	color: #776300;
}

.custom-block.danger {
	background-color: #ffe6e6 !important;
	border-color: #c00 !important;
}

.custom-block.danger .custom-block-title {
	color: #4d0000;
}
