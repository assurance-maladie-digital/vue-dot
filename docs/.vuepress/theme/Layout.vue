<template>
	<div class="theme-container">
		<VApp>
			<VToolbar
				app
				dark
				fixed
				clipped-left
				color="secondary"
			>
				<VBtn
					icon
					class="mr-3"
					@click="sidebarShow = !sidebarShow"
				>
					<SvgIcon icon="menu" />
				</VBtn>

				<VBtn
					flat
					:to="$t('home')"
					active-class="active"
					class="px-3 ma-0 text-none"
				>
					<SvgIcon
						x-large
						icon="vue-dot-dark"
					/>
					<h1 class="ml-2 mr-2 font-weight-regular">{{ $t('name') }}</h1>
				</VBtn>

				<VSpacer />

				<VTextField
					solo
					light
					:label="$t('toolbar.search')"
					single-line
					hide-details
					class="search mr-3"
					append-icon="search"
				/>

				<VMenu
					left
					bottom
					offset-y
				>
					<VBtn
						flat
						slot="activator"
						class="px-3 ma-0 text-none"
					>
						{{ $t('toolbar.ecosystem') }}
						<SvgIcon
							size=".7em"
							icon="down-arrow"
							class="ml-2"
						/>
					</VBtn>

					<VList>
						<VListTile
							v-for="item in $t('toolbar.items')"
							:key="item.title"
							:href="item.href"
							target="_blank"
							rel="noopener noreferrer"
							ripple
						>
							<VListTileContent>
								<VListTileTitle>
									<SvgIcon
										v-if="item.icon"
										:icon="item.icon"
										class="mr-2"
										color="#888"
									/>

									<span>{{ item.title }}</span>
								</VListTileTitle>
							</VListTileContent>
						</VListTile>
					</VList>
				</VMenu>

				<VBtn
					flat
					class="px-3 ma-0 text-none"
				>
					v{{ version }}
				</VBtn>

				<LangBtn
					flat
					flags
					menu-left
					menu-bottom
					:outline="false"
					:value="$i18n.locale"
					@change="updateLang"
					:display-text-btn="false"
					:display-arrow="false"
					:available-languages="languages"
					class="ma-0 text-none lang-btn"
					flags-url="https://res.cloudinary.com/deraw/image/upload/v1547044454/"
				/>
			</VToolbar>

			<VNavigationDrawer
				app
				fixed
				clipped
				:value="sidebar"
				v-model="sidebarShow"
			>
				<VList expand>
					<template v-for="item in $t('toolbar.sidebarItems')">
						<VListTile
							v-if="!item.items"
							:key="item.title"
							:to="item.to || null"
							:href="item.href || null"
							:target="item.href ? '_blank' : null"
							:rel="item.href ? 'noopener noreferrer' : null"
							ripple
							exact
						>
							<VListTileContent>
								<VListTileTitle>
									<SvgIcon
										v-if="item.icon"
										:icon="item.icon"
										class="mr-3"
									/>

									<span>{{ item.title }}</span>
								</VListTileTitle>
							</VListTileContent>
						</VListTile>

						<VListGroup
							v-if="item.items"
							:group="item.group"
							no-action
						>
							<VListTile
								:key="item.title"
								:to="item.to"
								slot="activator"
								ripple
								exact
							>
								<VListTileContent>
									<VListTileTitle>
										<SvgIcon
											v-if="item.icon"
											:icon="item.icon"
											class="mr-3"
										/>

										<span>{{ item.title }}</span>
									</VListTileTitle>
								</VListTileContent>
							</VListTile>

							<VListTile
								v-for="item in item.items"
								:key="item.title"
								:to="item.to"
								ripple
								exact
							>
								<VListTileContent>
									<VListTileTitle>
										<span>{{ item.title }}</span>
									</VListTileTitle>
								</VListTileContent>
							</VListTile>
						</VListGroup>
					</template>
				</VList>
			</VNavigationDrawer>

			<VContent>
				<component :is="layout" />
			</VContent>

			<VFooter
				v-if="$page.frontmatter.home"
				app
				dark
				absolute
				height="auto"
				class="px-3 py-5"
			>
				<VLayout
					column
					align-center
					justify-center
					class="footer-content"
				>
					<img
						:src="$withBase('/img/logo-am.svg')"
						height="70"
					>

					<p class="mt-3 mb-0">
						{{ $page.frontmatter.footerRealease }}
						<a href="https://github.com/assurance-maladie-digital/vue-dot/blob/master/LICENSE">
							{{ $page.frontmatter.footerRealeaseLink }}
						</a>
					</p>
					<p class="mb-0">
						{{ $page.frontmatter.footerMadeWith }}
						<a href="https://fr.vuejs.org/">
							{{ $page.frontmatter.footerVueJs }}
						</a>
					</p>
					<p class="mb-0">{{ $page.frontmatter.footerCopyright }}</p>
				</VLayout>
			</VFooter>
		</VApp>
	</div>
</template>

<script>
	import Home from './layouts/Home.vue';
	import Basic from './layouts/Basic.vue';
	import { version } from '../../../package.json';
	import { loadLanguageAsync } from '../i18n';

	export default {
		name: 'Layout',
		components: {
			Home,
			Basic
		},
		data() {
			return {
				version,
				initiated: false,
				languages: ['en', 'fr'],
				sidebarShow: false
			};
		},
		computed: {
			layout() {
				if (this.$page.frontmatter.home) {
					return 'Home';
				}

				return this.$page.frontmatter.layout || 'Basic'
			},
			sidebar() {
				return this.sidebarShow;
			}
		},
		methods: {
			updateLang(lang) {
				loadLanguageAsync(lang);

				if (this.initiated) {
					this.$router.push(`/${lang}/`);
				} else {
					this.initiated = true;
				}
			},
			guard() {
				// If the route doesn't match selected langage, redirect
				if (!this.$route.path.includes(this.$i18n.locale)) {
					this.$router.push(`/${this.$i18n.locale}/`);
				}
			}
		},
		watch: {
			$page() {
				this.sidebarShow = !this.$page.frontmatter.home;
			}
		},
		created() {
			if (!this.$vuetify.breakpoint.xsOnly) {
				this.sidebarShow = !this.$page.frontmatter.home;
			}

			// Redirect default route
			if (this.$route.path === '/') {
				this.$router.push(`/${this.$i18n.locale}/`);
			}

			this.guard();
		}
	}
</script>

<style scoped>
	.v-toolbar {
		z-index: 5 !important;
	}

	.v-toolbar .v-btn:not(.v-btn--icon),
	.v-menu,
	.v-menu >>> .v-menu__activator {
		height: 100%;
	}

	.footer-content {
		opacity: .9;
	}

	.v-footer a {
		color: #bbb;
	}

	.v-list__tile__title {
		display: flex;
		align-items: center;
	}

	.v-navigation-drawer >>> .v-list .v-list__tile {
		transition: background .15s !important;
	}

	.v-input.search {
		flex: none;
		height: 38px;
		width: 250px;
		overflow: hidden;
		border-radius: 50px;
	}

	.v-input.search >>> .v-input__control {
		min-height: 38px !important;
	}

	.lang-btn >>> .v-btn {
		margin: 0;
		height: 100%;
		padding: 20px;
		min-width: 50px;
	}
</style>

<style lang="stylus">
	@require '~vuetify/src/stylus/main'
	@require '../../../dist/vue-dot.css'

	@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons')
</style>
