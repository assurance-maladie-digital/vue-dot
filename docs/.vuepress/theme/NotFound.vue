<template>
	<VApp>
		<Toolbar />

		<VContent>
			<VContainer>
				 <VLayout align-center>
					<VFlex text-xs-center>
						<h1 class="display-2 primary--text mt-5 mb-3">{{ $t('error404.title') }}</h1>

						<p>{{ $t('error404.description') }}</p>

						<VBtn
							:to="`/${this.$i18n.locale}/guide/`"
							color="primary"
							outline
							exact
						>
							{{ $t('error404.button') }}
						</VBtn>
					</VFlex>
				</VLayout>
			</VContainer>
		</VContent>
	</VApp>
</template>

<script>
	import Toolbar from '../components/Toolbar.vue';

	export default {
		name: 'NotFound',
		components: {
			Toolbar
		},
		mounted() {
			// Redirect default route
			if (this.$route.path === '/') {
				this.$router.push(`/${this.$i18n.locale}/`);
			}
		}
	};
</script>
