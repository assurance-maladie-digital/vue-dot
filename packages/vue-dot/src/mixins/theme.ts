import Vue from 'vue';

const theme = {
	created(this: Vue) {
		if (this.$theme && this.$theme.config.colors) {
			this.$vuetify.theme = this.$theme.config.colors;
		}
	}
};

export default theme;
