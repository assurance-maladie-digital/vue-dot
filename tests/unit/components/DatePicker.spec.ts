import Vue from 'vue';
import { mount, createLocalVue } from '@vue/test-utils';
import DatePicker from 'vue-dot-src/components/DatePicker.vue';

import Vuetify from 'vuetify';
import VueDot from 'vue-dot-src/index';
import { default as theme } from '../theme.json';

const localVue = createLocalVue();
Vue.use(Vuetify);

localVue.use(VueDot, {
	theme
});

describe('DatePicker.vue', () => {
	const build = (locale: string) => {
		const wrapper = mount(DatePicker, {
			localVue,
			propsData: {
				locale
			}
		});

		const app = document.createElement('div');
		app.setAttribute('class', 'activator-icon-menu-picker-ref');
		document.body.append(app);

		return wrapper;
	};

	it('renders correctly', () => {
		// Test long locale
		const wrapper = build('pt-br');

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('cumputea a date', () => {
		const wrapper = build('fr');

		(wrapper.vm as any).date = '11-01-2001';
		expect((wrapper.vm as any).computedDateFormatted).toBe('01/11/2001');
	});

	it('handles a non-valid date', () => {
		const wrapper = build('fr');

		(wrapper.vm as any).date = 'not a valid date';
		wrapper.find('.v-text-field input').trigger('blur');
		wrapper.emitted('change');

		expect((wrapper.vm as any).date).toBe(null);
	});

	it('computes the date when the input is blurred', () => {
		const wrapper = build('fr-Fr');

		// Non valid date format
		(wrapper.vm as any).date = '11/01/2001';
		wrapper.find('.v-text-field input').trigger('blur');

		expect((wrapper.vm as any).date).toBe('2001-11-01');
	});

	it('handles a empty date', () => {
		const wrapper = build('fr');
		wrapper.setProps({
			locale: 'en'
		});

		(wrapper.vm as any).date = '';
		wrapper.find('.v-text-field input').trigger('blur');

		expect((wrapper.vm as any).date).toBe(null);
	});

	it('works with a mask', () => {
		const wrapper = build('fr');
		wrapper.setProps({
			mask: 'XX-XX-XXX'
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('works with a custom icon', () => {
		const wrapper = build('fr');
		wrapper.setProps({
			prependIcon: 'home'
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('works when the menu is activated', () => {
		const wrapper = build('fr');

		(wrapper.vm as any).menu = true;

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('works as a birthdate picker', () => {
		const wrapper = build('fr');

		expect((wrapper.vm as any).dateFormatted).toEqual('');

		wrapper.setProps({
			birthdate: true
		});

		(wrapper.vm as any).menu = true;
		(wrapper.vm as any).save();

		expect(wrapper.html()).toMatchSnapshot();
	});
});
