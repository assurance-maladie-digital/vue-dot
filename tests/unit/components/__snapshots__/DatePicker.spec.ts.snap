// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker.vue renders correctly 1`] = `
<div>
  <div class="v-input v-text-field theme--light">
    <div class="v-input__prepend-outer"><button type="button" class="ma-0 v-btn v-btn--icon theme--light activator-icon-menu-picker-ref" aria-label="Sélection de la date">
        <div class="v-btn__content"><span aria-hidden="true" class="x-svg-icon custom-color" style="color: rgb(128, 128, 128);"><svg viewBox="0 0 24 24"><g><path d="M12 13.2h6v6h-6z"></path> <path d="M20.4 2.4h-1.2V0h-2.4v2.4H7.2V0H4.8v2.4H3.6a2.39 2.39 0 0 0-2.39 2.4L1.2 21.6A2.4 2.4 0 0 0 3.6 24h16.8a2.4 2.4 0 0 0 2.4-2.4V4.8a2.4 2.4 0 0 0-2.4-2.4zm0 19.2H3.6V8.4h16.8z"></path></g></svg> </span></div>
      </button></div>
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: absolute;">Date</label><input aria-label="Date" type="text" maxlength="10"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <div class="v-messages__wrapper">
            <div class="v-messages__message">Format JJ/MM/AAAA</div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-menu">
    <div class="v-menu__content theme--light " style="max-height: auto; min-width: 300px; max-width: auto; top: 55px; left: 40px; z-index: 1; display: none;"></div>
  </div>
</div>
`;

exports[`DatePicker.vue works as a birthdate picker 1`] = `
<div>
  <div class="v-input v-text-field theme--light">
    <div class="v-input__prepend-outer"><button type="button" class="ma-0 v-btn v-btn--icon theme--light activator-icon-menu-picker-ref" aria-label="Sélection de la date">
        <div class="v-btn__content"><span aria-hidden="true" class="x-svg-icon custom-color" style="color: rgb(128, 128, 128);"><svg viewBox="0 0 24 24"><g><path d="M12 13.2h6v6h-6z"></path> <path d="M20.4 2.4h-1.2V0h-2.4v2.4H7.2V0H4.8v2.4H3.6a2.39 2.39 0 0 0-2.39 2.4L1.2 21.6A2.4 2.4 0 0 0 3.6 24h16.8a2.4 2.4 0 0 0 2.4-2.4V4.8a2.4 2.4 0 0 0-2.4-2.4zm0 19.2H3.6V8.4h16.8z"></path></g></svg> </span></div>
      </button></div>
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: absolute;">Date</label><input aria-label="Date" type="text" maxlength="10"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <div class="v-messages__wrapper">
            <div class="v-messages__message">Format JJ/MM/AAAA</div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-menu">
    <div class="v-menu__content theme--light menuable__content__active " style="max-height: auto; min-width: 300px; max-width: auto; top: 55px; left: 40px; z-index: 1; display: none;">
      <div class="v-picker v-card v-picker--date theme--light">
        <div class="v-picker__body theme--light" style="width: 300px;">
          <div>
            <div class="v-date-picker-header theme--light"><button type="button" class="v-btn v-btn--icon theme--light">
                <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_left</i></div>
              </button>
              <div class="v-date-picker-header__value">
                <div class="primary--text"><button type="button">2019 M02</button></div>
              </div><button disabled="disabled" type="button" class="v-btn v-btn--disabled v-btn--icon theme--light">
                <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_right</i></div>
              </button>
            </div>
            <div class="v-date-picker-table v-date-picker-table--date theme--light">
              <table>
                <thead>
                  <tr>
                    <th>M</th>
                    <th>T</th>
                    <th>W</th>
                    <th>T</th>
                    <th>F</th>
                    <th>S</th>
                    <th>S</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td></td>
                    <td></td>
                    <td></td>
                    <td></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">1</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">2</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">3</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">4</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">5</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">6</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">7</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--outline theme--light primary--text">
                        <div class="v-btn__content">8</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">9</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">10</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">11</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">12</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">13</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">14</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">15</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">16</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">17</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">18</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">19</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">20</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">21</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">22</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">23</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">24</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">25</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">26</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">27</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--disabled theme--light" disabled="">
                        <div class="v-btn__content">28</div>
                      </button></td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker.vue works when the menu is activated 1`] = `
<div>
  <div class="v-input v-text-field theme--light">
    <div class="v-input__prepend-outer"><button type="button" class="ma-0 v-btn v-btn--icon theme--light activator-icon-menu-picker-ref" aria-label="Sélection de la date">
        <div class="v-btn__content"><span aria-hidden="true" class="x-svg-icon custom-color" style="color: rgb(128, 128, 128);"><svg viewBox="0 0 24 24"><g><path d="M12 13.2h6v6h-6z"></path> <path d="M20.4 2.4h-1.2V0h-2.4v2.4H7.2V0H4.8v2.4H3.6a2.39 2.39 0 0 0-2.39 2.4L1.2 21.6A2.4 2.4 0 0 0 3.6 24h16.8a2.4 2.4 0 0 0 2.4-2.4V4.8a2.4 2.4 0 0 0-2.4-2.4zm0 19.2H3.6V8.4h16.8z"></path></g></svg> </span></div>
      </button></div>
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: absolute;">Date</label><input aria-label="Date" type="text" maxlength="10"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <div class="v-messages__wrapper">
            <div class="v-messages__message">Format JJ/MM/AAAA</div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-menu">
    <div class="v-menu__content theme--light menuable__content__active " style="max-height: auto; min-width: 300px; max-width: auto; top: 55px; left: 40px; z-index: 1; display: none;">
      <div class="v-picker v-card v-picker--date theme--light">
        <div class="v-picker__body theme--light" style="width: 300px;">
          <div>
            <div class="v-date-picker-header theme--light"><button type="button" class="v-btn v-btn--icon theme--light">
                <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_left</i></div>
              </button>
              <div class="v-date-picker-header__value">
                <div class="primary--text"><button type="button">2019 M02</button></div>
              </div><button type="button" class="v-btn v-btn--icon theme--light">
                <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_right</i></div>
              </button>
            </div>
            <div class="v-date-picker-table v-date-picker-table--date theme--light">
              <table>
                <thead>
                  <tr>
                    <th>M</th>
                    <th>T</th>
                    <th>W</th>
                    <th>T</th>
                    <th>F</th>
                    <th>S</th>
                    <th>S</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td></td>
                    <td></td>
                    <td></td>
                    <td></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">1</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">2</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">3</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">4</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">5</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">6</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">7</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating v-btn--outline theme--light primary--text">
                        <div class="v-btn__content">8</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">9</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">10</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">11</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">12</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">13</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">14</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">15</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">16</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">17</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">18</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">19</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">20</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">21</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">22</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">23</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">24</div>
                      </button></td>
                  </tr>
                  <tr>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">25</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">26</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">27</div>
                      </button></td>
                    <td><button type="button" class="v-btn v-btn--flat v-btn--floating theme--light">
                        <div class="v-btn__content">28</div>
                      </button></td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker.vue works with a custom icon 1`] = `
<div>
  <div class="v-input v-text-field theme--light">
    <div class="v-input__prepend-outer"><span aria-hidden="true" class="x-svg-icon custom-color activator-icon-menu-picker-ref" style="color: rgb(128, 128, 128);"><svg viewBox="0 0 24 24"><path d="M12 1.8L0 12.6h3.6v9.6h6V15h4.8v7.2h6v-9.6H24z"></path></svg> </span></div>
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: absolute;">Date</label><input aria-label="Date" type="text" maxlength="10"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <div class="v-messages__wrapper">
            <div class="v-messages__message">Format JJ/MM/AAAA</div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-menu">
    <div class="v-menu__content theme--light " style="max-height: auto; min-width: 300px; max-width: auto; top: 55px; left: 40px; z-index: 1; display: none;"></div>
  </div>
</div>
`;

exports[`DatePicker.vue works with a mask 1`] = `
<div>
  <div class="v-input v-text-field theme--light">
    <div class="v-input__prepend-outer"><button type="button" class="ma-0 v-btn v-btn--icon theme--light activator-icon-menu-picker-ref" aria-label="Sélection de la date">
        <div class="v-btn__content"><span aria-hidden="true" class="x-svg-icon custom-color" style="color: rgb(128, 128, 128);"><svg viewBox="0 0 24 24"><g><path d="M12 13.2h6v6h-6z"></path> <path d="M20.4 2.4h-1.2V0h-2.4v2.4H7.2V0H4.8v2.4H3.6a2.39 2.39 0 0 0-2.39 2.4L1.2 21.6A2.4 2.4 0 0 0 3.6 24h16.8a2.4 2.4 0 0 0 2.4-2.4V4.8a2.4 2.4 0 0 0-2.4-2.4zm0 19.2H3.6V8.4h16.8z"></path></g></svg> </span></div>
      </button></div>
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: absolute;">Date</label><input aria-label="Date" type="text" maxlength="9"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <div class="v-messages__wrapper">
            <div class="v-messages__message">Format JJ/MM/AAAA</div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-menu">
    <div class="v-menu__content theme--light " style="max-height: auto; min-width: 300px; max-width: auto; top: 55px; left: 40px; z-index: 1; display: none;"></div>
  </div>
</div>
`;
